### ISSUE TITLE
Intermittent Azure AD B2C exceptions: IDX10614: AsymmetricSecurityKey.GetSignatureFormater() throws an exception

### ISSUE THREAD

**Issue by bojanmisic** [azure-ad-b2c/samples#130]
We're using AAD B2C with custom policies for some time now, and everything worked as it should, but all of a sudden we started getting exceptions on sign-in policy. To make things worse, sometimes it does work but 4 out of 5 times we get an exception.

We managed to dig the error by linking policies to Application Insights, and here's what we got:

```
"Kind": "FatalException",
    "Content": {
      "Time": "9:05 PM",
      "Exception": {
        "Kind": "Handled",
        "HResult": "80131509",
        "Message": "IDX10614: AsymmetricSecurityKey.GetSignatureFormater( 'http://www.w3.org/2001/04/xmldsig-more#rsa-sha256' ) threw an exception.\nKey: 'System.IdentityModel.Tokens.X509AsymmetricSecurityKey'\nSignatureAlgorithm: 'http://www.w3.org/2001/04/xmldsig-more#rsa-sha256', check to make sure the SignatureAlgorithm is supported.\nException:'System.Security.Cryptography.CryptographicException: Invalid provider type specified.\r\n\r\n   at System.Security.Cryptography.Utils.CreateProvHandle(CspParameters parameters, Boolean randomKeyContainer)\r\n   at System.Security.Cryptography.Utils.GetKeyPairHelper(CspAlgorithmType keyType, CspParameters parameters, Boolean randomKeyContainer, Int32 dwKeySize, SafeProvHandle& safeProvHandle, SafeKeyHandle& safeKeyHandle)\r\n   at System.Security.Cryptography.RSACryptoServiceProvider.GetKeyPair()\r\n   at System.Security.Cryptography.RSACryptoServiceProvider..ctor(Int32 dwKeySize, CspParameters parameters, Boolean useDefaultKeySize)\r\n   at System.Security.Cryptography.X509Certificates.X509Certificate2.get_PrivateKey()\r\n   at System.IdentityModel.Tokens.X509AsymmetricSecurityKey.get_PrivateKey()\r\n   at System.IdentityModel.Tokens.X509AsymmetricSecurityKey.GetSignatureFormatter(String algorithm)\r\n   at System.IdentityModel.Tokens.AsymmetricSignatureProvider..ctor(AsymmetricSecurityKey key, String algorithm, Boolean willCreateSignatures)'.\nIf you only need to verify signatures the parameter 'willBeUseForSigning' should be false if the private key is not be available.",
        "Data": {},
        "Exception": {
          "Kind": "Handled",
          "HResult": "80090014",
          "Message": "Invalid provider type specified.\r\n",
          "Data": {}
        }
      }
    }
```

I have recreated all the policy keys of interest (B2C_1A_TokenEncryptionKeyContainer, B2C_1A_TokenSigningKeyContainer), but to no avail.

Also, incognito mode on web browser does not help.

Our sign-in policy is based on this example: [mfa-email-or-phone](https://github.com/azure-ad-b2c/samples/tree/master/policies/mfa-email-or-phone), with some custom additions.

Any clues appreciated - I can copy/paste traces from Application Insights if necessary or custom policy definitions.

**Comment by bojanmisic** [comment]
It's not intermittent any more, it happens always.

**Comment by bojanmisic** [comment]
I managed to fix the issue by using information from this thread: [stackoverflow](https://stackoverflow.com/questions/52063153/invalid-provider-type-specified-one-more-time).

Since we were also using [Invite Flow](https://github.com/azure-ad-b2c/samples/tree/master/policies/invite), I used the `New-SelfSignedCertificate` command noted there.

Now I have regenerated the certificate by using information from this [Microsoft doc](https://docs.microsoft.com/en-us/azure/active-directory-b2c/connect-with-saml-service-providers) and appending `Provider` parameter from that StackOverflow thread above:

```
PS C:\WINDOWS\system32> New-SelfSignedCertificate `
>>     -KeyExportPolicy Exportable `
>>     -Subject "CN=***.onmicrosoft.com" `
>>     -KeyAlgorithm RSA `
>>     -KeyLength 2048 `
>>     -KeyUsage DigitalSignature `
>>     -NotAfter (Get-Date).AddMonths(24) `
>>     -CertStoreLocation "Cert:\CurrentUser\My" `
>>     -Provider "microsoft enhanced rsa and aes cryptographic provider"
```

Although I'm not sure why this suddenly stopped working out of nowhere, I guess it could be that Microsoft updated something on their end, but again, not sure how only we had this problem :).

Anyways, it seems it works now, so we'll see.
